### ISSUE TITLE
在Aistudio中基于paddlehub调优问题

### ISSUE THREAD

**Issue by mrhan36** [PaddlePaddle/PaddleFormers#614]
您好，在Aistudio中基于paddlehub1.6进行优化调参，存在以下困惑，请您帮忙解答，谢谢！

## 问题1
如果代码如下：
input_dict, output_dict, program = module.context(trainable=False)

那是不是意味着除了softmax层，其他层都是无法finetune的，是不是就意味着以下代码就无效了？
"discriminative": {
"blocks": 3,
"factor": 2.6
},
"gradual_unfreeze":
{
"blocks": 3
}

##  问题2：
如果gradual_unfreeze的blocks=3，那么 "discriminative"的blocks取值是不是必须也是3，discriminitive_blocks的取值必须和gradual_unfreeze的blocks数值一致吗？是否允许以下情况，一个设置为3，一个设置为5？

"discriminative": {
"blocks": 5,
"factor": 2.6
},
"gradual_unfreeze":
{
"blocks": 3
}

##  问题3：
在Aistudio环境中，是否可使用paddlehub进行多卡并行训练？我使用以下代码进行多卡训练，似乎无效。
%set_env CUDA_VISIBLE_DEVICES=4
!export CUDA_VISIBLE_DEVICES=4
config = hub.RunConfig(
log_interval=10,
use_cuda=True,
num_epoch=20,
batch_size=32,
eval_interval=50,
use_data_parallel=True
)

##  问题4：
感觉slanted_triangle和wramup思路一致，在参数设置时，二者可以同时设置吗（如下代码），感觉这样设置冲突了，是否二选一即可？
scheduler = {
"warmup": 0.1,
"linear_decay": {
"start_point": 0.1,
"end_learning_rate": 0.0
},
"slanted_triangle": {
"cut_fraction": 0.1,
"ratio": 32
}

## 谢谢！

**Comment by kinghuin** [comment]
Hi，欢迎反馈问题

1. trainable=False会将预训练网络中的参数全部冻结，discriminative仍然会分块并设置学习率，但是已冻结的参数虽然有学习率但不会更新，由于预训练网络的参数通常远大于fc-softmax层，如果block没有设置得特别大的话，fc-softmax层通常都会是最后一个block，而最后一个block的学习率不会衰减。gradual_unfreeze则会将冻结的参数进行逐层解冻，使他们能够更新数值。不建议在trainable=False中使用这些策略。
2. 可以一个设置为3，一个设置为5
3. studio中只有单卡
4. 是的，slanted_triangle和wramup思路相似，同时设置它们会有warning，提示只有slanted_triangle生效。

**Comment by mrhan36** [comment]
> 1. 不会衰减

## 明白了，感谢回答！
